dataflow/src/org/checkerframework/dataflow/cfg/CFGBuilder.java

```diff
@@ -3397,19 +3402,42 @@ public Node visitConditionalExpression(ConditionalExpressionTree tree, Void p) {
             return node;
         }
 
+        private void extendsFinalliesUntil(Label l) {
+            assert finalliesLabelsStack.contains(l);
+
+            Deque<Object> oldFinalliesLabelsStack = finalliesLabelsStack;
+            finalliesLabelsStack = new LinkedList<>(finalliesLabelsStack);
+            while (!finalliesLabelsStack.isEmpty()) {
+                Object o = finalliesLabelsStack.pop();
+                if (l.equals(o)) {
+                    break;
+                } else if (o instanceof BlockTree) {
+                    BlockTree finallyBlock = (BlockTree) o;
```

Comment:
Some comment for what happens here would also be good. In particular, what happens with the copied tree?